### ISSUE TITLE
[a11y] spatial bookmarks

### ISSUE THREAD

**Issue by transkamp** [open-pioneer/trails-openlayers-base-packages#437]
### Issue description

open https://open-pioneer.github.io/trails-demo/openlayers-base-packages/dev/samples/showcase/?demo=spatialBookmarks

Nested interactive controls are not announced by screen readers
https://dequeuniversity.com/rules/axe/4.4/nested-interactive?application=axeAPI

![Image](https://github.com/user-attachments/assets/5775632a-b229-4bd2-b090-6f9bd45c3ae4)

chatgpt zeigt eine lösung auf wie man das problem beheben kann. in der linken spalte kann man alte chats abrufen und eben diese auch entfernen. bei betrachtung des HTML DOMs kann man die lösung erkennen
![Image](https://github.com/user-attachments/assets/c092a8a9-4857-4b94-98dd-3444969483eb)

########################

delete button within a list element requires more context per aria-label. it is important which bookmark exactly is deleted. therefore the title of the bookmark element must be linked to the delete button

![Image](https://github.com/user-attachments/assets/e2f25772-7dac-4202-a44f-cb8aa60a0568)

#################
a separator or a zebra layout between the individual list elements increases readability

### Validations

- [x] Read and follow our [code of conduct](https://github.com/open-pioneer/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [contributing guidelines](https://github.com/open-pioneer/trails-starter/blob/main/docs/Contributing.md)
- [x] Check that there isn't already a similar issue

**Comment by arnevogt** [comment]
PR #455 

PR implements a workaround for nested interactive controls.  

Findings:

- ChatGPT page uses an `<a>` tag with nested button instead of `<li>` with nested button
  - works better with screenreaders
  - though does not support navigation with up and down arrow keys
  
<details>

<summary>DOM example from chat gpt page</summary>

```html
<a
  tabindex="0"
  data-fill=""
  class="group __menu-item justify-between gap-6 data-fill:gap-2"
  draggable="true"
  href="/c/c916c678-80e4-49c5-90de-e14feb7ab4df"
  data-discover="true"
>
  <div class="flex min-w-0 grow items-center gap-2">
    <div class="truncate">
      <span class="" dir="auto">
        SPARQL Join Example <!-- comment: this is the title of the chat gpt conversation -->
      </span>
    </div>
  </div>
  <div class="text-token-text-tertiary flex items-center self-stretch">
    <div class="trailing highlight">
      <button
        tabindex="0"
        data-trailing-button=""
        class="__menu-item-trailing-btn"
        data-testid="history-item-0-options"
        aria-label="Open conversation options"
        type="button"
        id="radix-«r6d»"
        aria-haspopup="menu"
        aria-expanded="false"
        data-state="closed"
      >
        <div>
          <svg
            width="24"
            height="24"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
            class="icon-md"
            aria-hidden="true"
          >
            <path
              fill-rule="evenodd"
              clip-rule="evenodd"
              d="M3 12C3 10.8954 3.89543 10 5 10C6.10457 10 7 10.8954 7 12C7 13.1046 6.10457 14 5 14C3.89543 14 3 13.1046 3 12ZM10 12C10 10.8954 10.8954 10 12 10C13.1046 10 14 10.8954 14 12C14 13.1046 13.1046 14 12 14C10.8954 14 10 13.1046 10 12ZM17 12C17 10.8954 17.8954 10 19 10C20.1046 10 21 10.8954 21 12C21 13.1046 20.1046 14 19 14C17.8954 14 17 13.1046 17 12Z"
              fill="currentColor"
            ></path>
          </svg>
        </div>
      </button>
    </div>
  </div>
</a>
```

</details>

Futher Links:
[issue description - Interactive controls must not have focusable descendants.](https://accessibilityinsights.io/info-examples/web/nested-interactive/)
[similar question on stackexchange - suggest grid pattern instead of listbox pattern](https://stackoverflow.com/questions/77749475/how-to-fix-a11y-interactive-controls-must-not-be-nested-error)  (requires redesign of the component)
